_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp

```diff
@@ -378,11 +378,19 @@ namespace UMC_AV1_DECODER
                 updated_refs = prevFrameDPB;
                 Repeat_show = 0;
             }
+            if(Curr_temp)
```

Comment:
can we use ```std::unique_ptr``` instead ?

Reply:
+1

And I am not 100% sure that you've fully addressed a leak issue. Actually, you delete some memory in a GetFrame() function which, from the name, is a function called regularly on each frame. So, can this happen that in the very end of decoding, on decoder reset, etc. this memory will still be allocated and not destroyed? IMHO, std::unique_ptr would be much better way to handle that and guarantee no-leak state.

Reply:
IMHO, 
1. AV1DecoderFrame inherits from RefCounter. So we may need not use std::unique_ptr/share_ptr. Just need increase reference count when assignment and decrease reference count at the end of the scope of the pointer.
2. Curr/Curr_temp/Poutput should not be initialized by new operator when AV1Decoder constructor is called because  they are just used to pointer to some AV1DecoderFrame object in dpb array.
https://github.com/Intel-Media-SDK/MediaSDK/blob/5ff3e4c1a8ac459f51ff36e7a23cc1a1c9087b5c/_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp#L40-L42
https://github.com/Intel-Media-SDK/MediaSDK/blob/5ff3e4c1a8ac459f51ff36e7a23cc1a1c9087b5c/_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp#L627-L630
3. When AV1Decoder is destroyed, AV1DecoderFrame object in dpb array will be freed. No memory leak.